components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -162,6 +164,34 @@ class GeckoEngine(
         return speculativeSession ?: GeckoEngineSession(runtime, private, defaultSettings, contextId)
     }
 
+    /**
+     * See [Engine.createSessionWithDisplayMode].
+     */
+    override fun createSessionWithDisplayMode(
```

Comment:
I think this could be an optional (nullable) parameter in `createSession()` and doesn't require a separate method.